src/EditorFeatures/CSharpTest/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs

```diff
@@ -619,12 +620,54 @@ int M(int i)
         return i switch // trailing switch
         {
             // leading label
-            1 => 4, // trailing body
+            // trailing label
+            1 => 4,// leading body
+                   // trailing body
             2 => 5,
             3 => 6,
-
             // leading next statement
-            _ => throw null, // leading next statement
+            _ => throw null,// leading next statement
+        };
+    }
+}");
+        }
+
+        [WorkItem(37873, "https://github.com/dotnet/roslyn/issues/37873")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertSwitchStatementToExpression)]
+        public async Task TestTrivia_02()
+        {
+            await TestInCSharp8(
+@"class Program
+{
+    static int GetValue(int input)
+    {
+        [||]switch (input)
+        {
+            case 1:
+                // this little piggy went to market
+                return 42;
+            case 2:
+                // this little piggy stayed home
+                return 50;
+            case 3:
+                // this little piggy had roast beef
+                return 79;
+            default:
+                // this little piggy had none
+                return 80;
+        }
+    }
+}",
+@"class Program
+{
+    static int GetValue(int input)
+    {
+        return input switch
+        {
+            1 => 42,// this little piggy went to market
```

Comment:
nit.  probably want space after the comma.

Reply:
(super super minor nit)

Reply:
that's just me not knowing how trivia works. there is no resource and I should just infer from code. not an easy task for something this subtle.